vulkan_generator/struct_factories/generator.py

```diff
@@ -0,0 +1,451 @@
+# Copyright (C) 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This is the entry point for Vulkan Code Generator"""
+
+import abc
+
+from pathlib import Path
+from typing import List
+from typing import Dict
+
+from textwrap import dedent
+from vulkan_generator.vulkan_parser import types
+from vulkan_generator.codegen_utils import codegen
+
+
+def struct_factory_name(struct: str) -> str:
+    return struct + "Factory"
+
+
+def set_member_name(member: str) -> str:
+    return "set" + member[0:1].upper() + member[1:]
```

Comment:
`member[0:1].upper() + member[1:]` this seems to be something that we can put in a function. It's a shame that Python doesn't have this option in capitlize()